packages/core/src/remoteconfig.ts

```diff
@@ -0,0 +1,211 @@
+  //   const lastFetched = storage.get(`${defaultConfigName}_lastFetch`);
+  // }
+
+  /**
+   * Loads a fetched/cached config so that it is usable by public APIs
+   */
+  function _loadConfig(config: RemoteConfigPayload): void {
+    _activeConfig = {
+      features: config.features.reduce<Record<string, any>>((acc, { key, value }) => {
+        acc[key] = value;
+        return acc;
+      }, {}),
+      options: config.options,
+      version: config.version,
+    };
+  }
+
+  /**
+   *
+   */
+  function _fetch(): PromiseLike<void> {
+    return new Promise((resolve, reject) => {
+      if (!_transport) {
+        return resolve();
+      }
+
+      // TODO
+      return (
+        _transport
+          // @ts-expect-error envelopes and such
+          .send([[], [[{ type: 'features' }, {}]]])
+          .then(resp => {
+            // _lastFetch = new Date();
+            // on success, check if cached, then do nothing
+            if (resp.statusCode === 304) {
+              // _state = "SUCCESS_CACHED";
+
+              return;
+            }
+
+            // not cached...get body and send pending
+            // @ts-expect-error resp.response not typed
+            resp.response
+              .json()
+              .then((data: RemoteConfigPayload) => {
+                // _lastFetch = new Date();
+                _pendingConfig = data;
+                _hasUpdate = true;
+
+                storage.set(defaultConfigName, data);
+                // storage.set(`${defaultConfigName}_lastFetch`, +_lastFetch);
+                // _state = "SUCCESS";
+                resolve();
+              })
+              .catch(() => {
+                // TODO: Error handling
+              });
+          })
+      );
+    });
+  }
+
+  /**
+   * Applies remote configuration updates to local configuration.
+   */
+  function applyUpdate(): boolean {
+    if (!_pendingConfig || !_hasUpdate) {
+      // Nothing to do;
+      return false;
+    }
+
+    _loadConfig(_pendingConfig);
+    _pendingConfig = undefined;
+    _source = 'REMOTE';
+    _lastUpdate = new Date();
+
+    return true;
+  }
+
+  /**
+   * Async function to fetch and apply configuration.
+   */
+  async function fetchAndApply(): Promise<void> {
+    await _fetch();
+    applyUpdate();
+  }
+
+  /**
+   * Fetch configuration, but does *not* apply. Call `applyConfig` to use fetched configuration.
+   */
+  function fetch(): void {
+    void _fetch();
+  }
+
+  /**
+   *
+   */
+  function getSource(): 'DEFAULT' | 'CACHED' | 'REMOTE' {
+    return _source;
+  }
+
+  /** @inheritDoc */
+  function getInternal(config: RemoteOverrideableConfig): RemoteOverrideableConfig {
+    return {
+      sampleRate: (_activeConfig && _activeConfig.options.sample_rate) || config.sampleRate,
+      tracesSampleRate: (_activeConfig && _activeConfig.options.traces_sample_rate) || config.tracesSampleRate,
+    };
+  }
+
+  /** @inheritdoc */
+  function get<T>(defaultConfig: T): T {
```

Comment:
It's a big cumbersome to pass in `defaultConfig` each time, unless there's something else for the public api.

Users could endup having to do:
```
const is_enabled = remoteConfigInst.get({is_enabled: false}).is_enabled;
```

Reply:
Ah yeah this hasn't been updated for the new API, will be: `get(key, defaultValue)`